cxbox-core/src/main/java/org/cxbox/core/dao/impl/MetadataUtils.java

```diff
@@ -207,37 +212,50 @@ public static Path getFieldPath(String fieldName, Root<?> root) {
 		return root.get(fieldName);
 	}
 
-	public static Predicate createPredicate(Root<?> root, ClassifyDataParameter criteria, CriteriaBuilder cb) {
+	public static Predicate createPredicate(Root<?> root, ClassifyDataParameter criteria, CriteriaBuilder cb,
+			String dialect, ApplicationContext applicationContext) {
 		try {
 			Object value = criteria.getValue();
 
 			Path field = getFieldPath(criteria.getField(), root);
 
-			switch (criteria.getOperator()) {
+			ClassifyDataProvider classifyDataProvider = getProviderFromParam(criteria.getProvider(), applicationContext);
+			Expression filterPredicate = classifyDataProvider.getFilterPredicate(cb, criteria, field, dialect, value);
```

Comment:
Method getFilterPredicate but returns expression - not predicate??

Return Predicate. Not expression. Add criteria.getOperator() as input param to getFilterPredicate. if (filterPredicate != null)  -- use predicate from classifyDataProvider. else - do default switch logic

Reply:
also add Root<?> root, ClassifyDataParameter criteria, CriteriaBuilder cb,
			String dialect params to prevent api change in future